src/dma/mod.rs

```diff
@@ -636,131 +683,40 @@ where
         $(,)*) => {
         $(
             impl<I: Instance> StreamISR for StreamX<I, $number> where Self: crate::Sealed {
-
-                #[inline(always)]
-                fn clear_all_flags(&mut self) {
-                    //NOTE(unsafe) Atomic write with no side-effects and we only access the bits
-                    // that belongs to the StreamX
-                    let dma = unsafe { &*I::ptr() };
-                    dma.$ifcr.write(|w| w
-                        .$tcif().set_bit() //Clear transfer complete interrupt flag
-                        .$htif().set_bit() //Clear half transfer interrupt flag
-                        .$teif().set_bit() //Clear transfer error interrupt flag
-                        .$dmeif().set_bit() //Clear direct mode error interrupt flag
-                        .$feif().set_bit() //Clear fifo error interrupt flag
-                    );
-                }
-
-                #[inline(always)]
-                fn clear_transfer_complete_flag(&mut self) {
-                    //NOTE(unsafe) Atomic write with no side-effects and we only access the bits
-                    // that belongs to the StreamX
-                    let dma = unsafe { &*I::ptr() };
-                    dma.$ifcr.write(|w| w.$tcif().set_bit());
-                }
-
                 #[inline(always)]
-                fn clear_half_transfer_flag(&mut self) {
-                    //NOTE(unsafe) Atomic write with no side-effects and we only access the bits
-                    // that belongs to the StreamX
+                fn clear_flags(&mut self, flags: DmaFlags) {
                     let dma = unsafe { &*I::ptr() };
-                    dma.$ifcr.write(|w| w.$htif().set_bit());
-                }
-
-                #[inline(always)]
-                fn clear_transfer_error_flag(&mut self) {
-                    //NOTE(unsafe) Atomic write with no side-effects and we only access the bits
-                    // that belongs to the StreamX
-                    let dma = unsafe { &*I::ptr() };
-                    dma.$ifcr.write(|w| w.$teif().set_bit());
-                }
-
-                #[inline(always)]
-                fn clear_direct_mode_error_flag(&mut self) {
-                    //NOTE(unsafe) Atomic write with no side-effects and we only access the bits
-                    // that belongs to the StreamX
-                    let dma = unsafe { &*I::ptr() };
-                    dma.$ifcr.write(|w| w.$dmeif().set_bit());
-                }
-
-                #[inline(always)]
-                fn clear_fifo_error_flag(&mut self) {
-                    //NOTE(unsafe) Atomic write with no side-effects and we only access the bits
-                    // that belongs to the StreamX
-                    let dma = unsafe { &*I::ptr() };
-                    dma.$ifcr.write(|w| w.$feif().set_bit());
+                    dma.$ifcr.write(|w| unsafe { w.bits(flags.bits() as u32) });
```

Comment:
same remark as listen you need to to delete bit that doesn't correspond to a flag. And I think you missed to apply $isr_shift

Reply:
missed this in previous commit